src/main/java/uk/nhs/tis/trainee/notifications/event/GmcUpdateListener.java

```diff
@@ -92,31 +81,7 @@ public void handleGmcUpdate(GmcUpdateEvent event) throws MessagingException {
     templateVariables.put(GMC_NUMBER_FIELD, event.gmcDetails().gmcNumber());
     templateVariables.put(GMC_STATUS_FIELD, event.gmcDetails().gmcStatus());
 
-    String traineeId = event.traineeId();
-    Set<LocalOfficeContact> localOfficeContacts = notificationService
-        .getTraineeLocalOfficeContacts(traineeId, GMC_UPDATE);
-
-    if (localOfficeContacts != null && !localOfficeContacts.isEmpty()) {
-      boolean canSendMail = messagingControllerService.isMessagingEnabled(MessageType.EMAIL);
-      //since some LO's share a contact we need to eliminate possible duplicates:
-      Set<String> distinctContacts = localOfficeContacts.stream()
-          .map(LocalOfficeContact::contact).filter(Objects::nonNull).collect(Collectors.toSet());
-
-      for (String loContact : distinctContacts) {
-        if (notificationService.isLocalOfficeContactEmail(loContact)) {
-          emailService.sendMessage(traineeId, loContact, GMC_UPDATED, templateVersion,
-              templateVariables, null, !canSendMail);
-          log.info("GMC updated notification {} for trainee {} to {}.",
-              (canSendMail ? "sent" : "logged"), traineeId, loContact);
-        } else {
-          log.info("GMC updated notification skipped for trainee {} to non-email {}.",
-              traineeId, loContact);
-        }
-      }
-
-    } else {
-      log.warn("GMC updated notification not processed for trainee {}: no local office contacts.",
-          traineeId);
-    }
+    notificationService.sendLocalOfficeMail(event.traineeId(), GMC_UPDATE, templateVariables,
+        templateVersion, GMC_UPDATED);
```

Comment:
This has been raised-up into the NotificationService, though I think that class is getting a bit unwieldly and is need of a bit or refactoring/rethinking.